src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -445,7 +445,7 @@ public function resolve($file)
     {
         $class = Str::studly(implode('_', array_slice(explode('_', $file), 4)));
 
-        return new $class;
+        return resolve($class);
```

Comment:
We can't use a global foundation helper here. This couples the database component directly to the framework and will prevent it from being used outside of it.